src/test/java/io/cdap/plugin/sfmc/sink/MarketingCloudDataExtensionSinkTest.java

```diff
@@ -0,0 +1,151 @@
+ * the License.
+ */
+package io.cdap.plugin.sfmc.sink;
+
+import com.exacttarget.fuelsdk.ETClient;
+import io.cdap.cdap.api.data.batch.OutputFormatProvider;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.batch.BatchRuntimeContext;
+import io.cdap.cdap.etl.api.batch.BatchSinkContext;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.mock.common.MockArguments;
+import io.cdap.cdap.etl.mock.common.MockPipelineConfigurer;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DataExtensionClient.class, ETClient.class})
+public class MarketingCloudDataExtensionSinkTest {
+  protected static final String CLIENT_ID = "clientId";
+  protected static final String CLIENT_SECRET = "clientSecret";
+  protected static final String AUTH_ENDPOINT = "authEndPoint";
+  protected static final String SOAP_ENDPOINT = "soapEndPoint";
+  private static final Logger LOG = LoggerFactory.getLogger(MarketingCloudDataExtensionSink.class);
+  private MarketingCloudConf marketingCloudConf;
+  private MarketingCloudDataExtensionSink marketingCloudDataExtensionSink;
+
+  @Before
+  public void initialize() {
+    marketingCloudConf = MarketingCloudConfHelper.newConfigBuilder().
+      setReferenceName("referenceName")
+      .setAuthEndpoint(AUTH_ENDPOINT)
+      .setClientId(CLIENT_ID)
+      .setClientSecret(CLIENT_SECRET)
+      .setSoapEndpoint(SOAP_ENDPOINT)
+      .setDataExtension("DE")
+      .setMaxBatchSize(500)
+      .setOperation("INSERT")
+      .setColumnMapping("<key=value>").
+      setFailOnError(false)
+      .setTruncateText(false)
+      .setReplaceWithSpaces(false)
+      .build();
+    marketingCloudDataExtensionSink = new MarketingCloudDataExtensionSink(marketingCloudConf);
+  }
+
+  @Test
+  public void testInitialize() {
+    BatchRuntimeContext context = Mockito.mock(BatchRuntimeContext.class);
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    Mockito.when(context.getFailureCollector()).thenReturn(failureCollector);
+    Schema fieldSchema = Schema.recordOf("record",
+                                         Schema.Field.of("store_id", Schema.of(Schema.Type.INT)),
+                                         Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)),
+                                         Schema.Field.of("emailid", Schema.of(Schema.Type.STRING)));
+    Mockito.when(context.getInputSchema()).thenReturn(fieldSchema);
+    marketingCloudDataExtensionSink.initialize(context);
```

Comment:
you're not validating anything in this testcase

Reply:
Removed this test case as it is simply used to initialize the BatchSink, so tested it along with other method.